frontend/packages/kubevirt-plugin/src/components/modals/vm-flavor-modal/vm-flavor-modal.tsx

```diff
@@ -1,154 +1,195 @@
 import * as React from 'react';
 import * as _ from 'lodash';
-import * as classNames from 'classnames';
+import { Form, FormSelect, FormSelectOption } from '@patternfly/react-core';
 import {
+  Firehose,
+  FirehoseResult,
   HandlePromiseProps,
+  validate,
   withHandlePromise,
-  Dropdown,
-  convertToBaseValue,
-  Firehose,
 } from '@console/internal/components/utils';
 import { TemplateModel } from '@console/internal/models';
 import {
   createModalLauncher,
-  ModalTitle,
   ModalBody,
   ModalComponentProps,
-  ModalFooter,
+  ModalTitle,
 } from '@console/internal/components/factory';
 import { k8sPatch, TemplateKind } from '@console/internal/module/k8s';
-import { Form, FormGroup, TextInput } from '@patternfly/react-core';
 import { VMLikeEntityKind } from '../../../types/vmLike';
 import {
+  asVM,
+  getCPU,
   getFlavor,
   getMemory,
-  getCPU,
-  vCPUCount,
-  asVM,
   getVMLikeModel,
+  vCPUCount,
 } from '../../../selectors/vm';
 import { getFlavors } from '../../../selectors/vm-template/selectors';
 import { getUpdateFlavorPatches } from '../../../k8s/patches/vm/vm-patches';
 import {
   CUSTOM_FLAVOR,
   NAMESPACE_OPENSHIFT,
-  TEMPLATE_TYPE_LABEL,
   TEMPLATE_TYPE_BASE,
+  TEMPLATE_TYPE_LABEL,
 } from '../../../constants';
-import { getResource } from '../../../utils';
-import './_vm-flavor-modal.scss';
-
-const Gi = 1024 ** 3;
+import { getLoadedData, getResource } from '../../../utils';
+import { SizeUnitFormRow } from '../../form/size-unit-form-row';
+import { BinaryUnit } from '../../form/size-unit-utils';
+import { ModalFooter } from '../modal/modal-footer';
+import { FormRow } from '../../form/form-row';
+import { Integer } from '../../form/integer/integer';
+import { validateFlavor } from '../../../utils/validations/vm/flavor';
+import { isValidationError } from '../../../utils/validations/common';
+import { useShowErrorToggler } from '../../../hooks/use-show-error-toggler';
+import { getDialogUIError } from '../../../utils/strings';
 
 const getId = (field: string) => `vm-flavor-modal-${field}`;
-const dehumanizeMemory = (memory?: string) => {
-  if (!memory) {
-    return null;
-  }
-
-  return convertToBaseValue(memory) / Gi;
-};
 
 const VMFlavorModal = withHandlePromise((props: VMFlavornModalProps) => {
   const {
     vmLike,
     templates,
-    inProgress,
     errorMessage,
     handlePromise,
     close,
     cancel,
     loadError,
     loaded,
   } = props;
+  const inProgress = props.inProgress || !loaded;
   const vm = asVM(vmLike);
-  const flattenTemplates = _.get(templates, 'data', []) as TemplateKind[];
+  const flattenTemplates = getLoadedData(templates, []);
 
   const vmFlavor = getFlavor(vmLike);
```

Comment:
When VM does not have a flavor `getFlavors` will default to `[custom_flavor]` but `getFlavor` will return `null`
So in cases VM is missing a floavor we will get one option 'custom', but `isCustom` will be false.

I think this is a nice work around:
```
const flavors = getFlavors(vmLike, flattenTemplates);
const vmFlavor = getFlavor(vmLike) || flavors[0];
```

Reply:
+1, that is much better, but using the last flavor due to the sorting